bindings/nodejs/src/connection.rs

```diff
@@ -1,131 +1,41 @@
-                // Assume failing to parse a url just means the user provided a local path.
-                Err(_) => JsSessionConf {
-                    data_dir: Some(PathBuf::from(s)),
-                    cloud_url: None,
-                },
-            },
-            None => JsSessionConf {
-                data_dir: None,
-                cloud_url: None,
-            },
-        }
-    }
+    pub(crate) inner: Arc<glaredb::Connection>,
 }
 
 #[napi]
 impl Connection {
-    pub(crate) async fn connect(
-        data_dir_or_cloud_url: Option<String>,
-        spill_path: Option<String>,
-        disable_tls: bool,
-        cloud_addr: String,
-        location: Option<String>,
-        storage_options: Option<HashMap<String, String>>,
-    ) -> napi::Result<Self> {
-        let conf = JsSessionConf::from(data_dir_or_cloud_url);
-
-        let storage = if let Some(location) = location.clone() {
-            EngineStorage::Remote {
-                location,
-                options: storage_options.unwrap_or_default(),
-            }
-        } else if let Some(data_dir) = conf.data_dir.clone() {
-            EngineStorage::Local(data_dir)
-        } else {
-            EngineStorage::Memory
-        };
-
-        let mut engine = Engine::from_storage(storage)
-            .await
-            .map_err(JsGlareDbError::from)?;
-
-        engine = engine
-            .with_spill_path(spill_path.map(|p| p.into()))
-            .map_err(JsGlareDbError::from)?;
-
-        let mut session = engine
-            .default_local_session_context()
-            .await
-            .map_err(JsGlareDbError::from)?;
-
-        session
-            .create_client_session(
-                conf.cloud_url.clone(),
-                cloud_addr,
-                disable_tls,
-                RemoteClientType::Node,
-                None,
-            )
-            .await
-            .map_err(JsGlareDbError::from)?;
-
-        Ok(Connection {
-            session: Arc::new(Mutex::new(session)),
-            _engine: Arc::new(engine),
-        })
-    }
-
     /// Returns the default connection to a global in-memory database.
     ///
     /// The database is only initialized once, and all subsequent
     /// calls will return the same connection object and therefore
     /// have access to the same data.
     #[napi(catch_unwind)]
     pub async fn default_in_memory() -> napi::Result<Connection> {
-        let engine = Engine::from_storage(EngineStorage::Memory)
-            .await
-            .map_err(JsGlareDbError::from)?;
-        let sess = engine
-            .new_local_session_context(SessionVars::default(), SessionStorageConfig::default())
-            .await
-            .map_err(JsGlareDbError::from)?;
-        let con = Connection {
-            session: Arc::new(Mutex::new(sess)),
-            _engine: Arc::new(engine),
-        };
-
-        Ok(con.clone())
+        static DEFAULT_CON: OnceCell<Connection> = OnceCell::new();
```

Comment:
we should probably just return a new instance each time. This seems like it could produce some unexpected behavior. 

such as. 

```js
const db1 = glaredb.connect() 

const db2 = glaredb.connect() 
```

I'd expect these to be two separate sessions, but this code looks like they'd use a shared connection.



JS isn't really known for it's performance, so the cost of spinning up a new session is likely not a concern. It's what we are already doing, and seems a bit safer a

Reply:
> I'd expect these to be two separate sessions, but this code looks like they'd use a shared connection.

This is complicated: 
- before https://github.com/GlareDB/glaredb/commit/138347b287c94d53cbe37ee3f90b9284a08c42ae, there was only one glaredb in memory database (particularly confusing for testing, I think, but also users.) 
- as of #2850, the python driver has (will have) the semantics you (and I) expect, as above.
- also as of that PR the `glaredb.sql()` and `glaredb.execute()` functions in python, will access this static/global shared db.
- I believe, that we could have a similar (global) functionality in node, but I'd be happy to drop it. my grep shows that this is never called.